content/realtime/channels/channel-parameters/rewind.textile

```diff
@@ -0,0 +1,87 @@
+
+At most 100 messages will be sent in a rewind request. If the number of messages within the specified interval is greater than that limit, then only the most recent messages up to that limit are sent. The attachment succeeds, but truncation of the message backlog is indicated as a non-fatal error in the attachment response.
+
+By default, a maximum of two minutes of channel history is available when attaching. This means that a rewind time specifier of greater than two minutes will only be able to rewind by two minutes. If a channel has persistence enabled, then it is possible to rewind back in time by up to the persistence TTL on the channel.
+
+The channel position expressed by a @rewind@ parameter only has an effect on an initial channel attachment. Any subsequent reattachment of the same channel on the same connection, in order to resume the connection, will attempt to resume with continuity from the point at which the connection dropped. (There are a few exceptions to this: in particular, client libraries earlier than v1.2 that have been disconnected for over two minutes, and all clients when using "@recover@ mode":/realtime/connection#connection-state-recovery ; in both cases the previous attachment state is not preserved).
+
+Any @rewind@ parameter value that cannot be parsed either as a number or a time specifier will cause the attachment request to fail and return an error.
+
+h2(#examples). Examples
+
+h3(#rewind-example-ably). Rewind example with an Ably client library
+
+To subscribe to a channel, getting the most recent message if available:
+
+```[jsall]
+  // only with ably-js v1.2 or later
+  const realtime = new Ably.Realtime('{{API_KEY}}');
+  realtime.channels.get('{{RANDOM_CHANNEL_NAME}}', {
+    rewind: '1'
+  }).subscribe(msg => console.log("Received message: ", msg));
+```
+
+```[jsall]
+  // with ably-js v1.1 or below
+  const realtime = new Ably.Realtime('{{API_KEY}}');
+  const channel = realtime.channels.get('[?rewind=1]{{RANDOM_CHANNEL_NAME}}');
+  channel.subscribe(msg => console.log("Received message: ", msg));
+```
+
+h3(#rewind-example-sse). Rewind example with SSE
+
+To subscribe to a channel, getting the most recent message if available:
+
+```[javascript]
+  var querystring = 'v=1.2&channels={{RANDOM_CHANNEL_NAME}}&rewind=1&key={{API_KEY}}';
+  var eventSource = new EventSource('https://realtime.ably.io/event-stream?' + querystring);
+```
+
+h3(#rewind-examples-mqtt). Rewind example with MQTT
+
+```[nodejs]
+  var mqtt = require('mqtt');
+  var options = {
+    keepalive: 30,
+    username: 'FIRST_HALF_OF_API_KEY',
```

Comment:
Why does this example not use an actual API key like all other examples?

Reply:
@mattheworiordan How would that look? What do you mean by "actual API key" in this context?

The MQTT library requires the 'split' key - key name and key secret. This example derives from my [internal prover](https://github.com/ably/hello-deltas-js/blob/8faff1cbb2c4c2a8325469511dadae12f1e3a733/server.js#L217).

Reply:
I don't think we use any actual keys in the docs other than JSBin examples, @tomczoink to confirm.

Reply:
same: https://github.com/ably/docs/pull/880/files#r433847971

Reply:
See https://docs.ably.io/client-lib-development-guide/documentation-formatting-guide/#code-blocks, section "Variables".

Specifically:

`{{API_KEY_NAME}}` and `{{API_KEY_SECRET}}`

The docs repo has a lot of engineering to bring our code samples to life, through injecting usable API keys, to launching live code samples.  This is one of those examples.  I highly recommend you read through https://docs.ably.io/client-lib-development-guide/documentation-formatting-guide/ in depth.